### ISSUE TITLE
Web interface: design/accessibility issues

### ISSUE THREAD

**Issue by ferdnyc** [openscanhub/openscanhub#214]
As I mentioned in a comment on #211, the current web reporting interface has several design issues. These include some severe accessibility concerns that would make the interface difficult to work with for the portion of the community who rely on assistive technologies and accessible design in order to comfortably browse the web.

I noticed issue #152 asking the question of whether the site should be designed using PatternFly — I don't have an opinion to offer on that tool _specifically_, but **something** should be used to ensure that the interface is designed in observance of basic accessibility criteria. The current design falls miles short.

Some of the issues:

1. The color of comment lines, in the source output, is `#00c0c0`, which against a white background has an abysmal contrast ratio of 2.25.<sup>1</sup>
2. The non-highlighted source lines fare even worse; rendered in `#c0c0c0` on solid white, the resulting contrast ratio is only 1.81.
3. Worst of all, **EVERY** piece of text is colored by wrapping it in a `<span>` element with a direct `style=` attribute, making it impossible to modify the page style using CSS.

The third item is the real four-alarm fire, because the first two items would at least be addressable using local CSS rules to change the page styling. Except, with only `<span style="color:#xxxxxx">`, there effectively **IS NO** page styling to change.

**Do. Not. Do. This.** It's almost 2024, and this _cannot_ be how web interfaces are built. Use semantic markup, apply CSS classes to page elements, and define style parameters using CSS rules applied to those classes.

Designed that way, even an accessibility disaster of a page can be improved with judicious application of a few local CSS overrides.

#### Notes
1. (For reference, the [Web Content Accessibility Guidelines (WCAG)](https://www.w3.org/TR/WCAG21/) provide a two-level set of criteria for color contrast. In body text, the WCAG AA level (considered the minimum standard for comfortable viewing of text) requires a contrast ratio of 4.5 or higher. The more stringent AAA level requires a minimum 7.0 contrast ratio.

**Comment by kdudka** [comment]
As pointed out by @lzaoral, the issues with HTML rendering of scan results need to be addressed in `csdiff`, which is a separate project that OpenScanHub uses internally.  This is the code that needs to be improved: https://github.com/csutils/csdiff/blob/df1fda07a6fd6bb81d717d253d170935feb0886a/src/lib/writer-html.cc

**Comment by siteshwar** [comment]
I am closing this issue, as it needs to be fixed in the [csdiff](https://github.com/csutils/csdiff/issues/174) code.
